src/neuro_relay/intermediary.py

```
@@ -134,26 +160,33 @@ async def _notify_watchers(self, message: dict):
+    ) -> None:
+        """Integration -> Relay
         Integration sends JSON or plain text messages. We forward to Neuro via on_forward_to_neuro if present.
         Expected integration message shape:
-            {"action":"name", "params": {...}, "meta": {...}}
+            {"action":"name", "params": {...}, "meta": {...}}.
         """
         async for raw in ws:
             # handle binary frames (e.g., file uploads)
             if isinstance(raw, (bytes, bytearray)):
                 filename = f"upload_{origin_name}.bin"
                 with open(filename, "wb") as f:
```

Ruff (0.14.8)

Async functions should not open files with blocking methods like `open`

(ASYNC230)